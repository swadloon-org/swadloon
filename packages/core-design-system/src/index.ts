export * from './brand/logos.js';
export * from './components/box.js';
export * from './components/buttons.js';
export * from './components/carousel.js';
export * from './components/components.js';
export * from './components/inputs.js';
export * from './components/links.js';
export * from './components/list-items.js';
export * from './components/navbar-item.js';
export * from './components/paragraphs.js';
export * from './components/size.js';
export * from './components/table-cells.js';
export * from './components/tabs.js';
export * from './components/tags.js';
export * from './design-system.js';
export * from './foundations/animations.js';
export * from './foundations/background-effect.js';
export * from './foundations/code-colors.js';
export * from './foundations/color-gradients.js';
export * from './foundations/color-intents.js';
export * from './foundations/color-scheme.js';
export * from './foundations/colors.js';
export * from './foundations/colors-colors.js';
export * from './foundations/content-width.js';
export * from './foundations/effects.js';
export * from './foundations/font.js';
export * from './foundations/iconography.js';
export * from './foundations/layout.js';
export * from './foundations/media-queries.js';
export * from './foundations/sizing.js';
export * from './foundations/sizing-modular-scale.js';
export * from './foundations/text.js';
export * from './foundations/typography.js';
export * from './primitives/angle.js';
export * from './primitives/color.js';
export * from './primitives/color-gradient.js';
export * from './primitives/color-palette.js';
export * from './primitives/radius.js';
export * from './types.js';
