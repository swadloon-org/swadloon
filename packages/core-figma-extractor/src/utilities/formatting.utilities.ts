import Case from 'case';
import prettier, { Options } from 'prettier';

const prettierConfig: Options = {
  singleQuote: true,
  tabWidth: 2,
  printWidth: 120,
};

/**
 * Format text content
 */
export function formatOutput(textContent: string, options: prettier.Options) {
  return prettier.format(textContent, { ...prettierConfig, ...options });
}

/**
 * Split numbers in camel case with '-'
 * @example info900 => info-900
 */
export function formatNameNumberKebabCase(text: string) {
  const kebabText = Case.kebab(text);
  const match = /[a-z](\d+)/gi.exec(kebabText);

  if (!match) {
    return kebabText;
  }

  const splittedText = [kebabText.slice(0, match.index + 1), kebabText.slice(match.index + 1)].join(
    '-'
  );
  return splittedText;
}
