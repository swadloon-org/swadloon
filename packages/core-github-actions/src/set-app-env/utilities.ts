import * as core from '@actions/core';

import { TEST_ENV } from '@newrade/core-common';

import { ENV } from '../../types/dot-env.js';

/**
 * Export an env variable using @actions/core's utility (env as any) for tests
 *
 * @param name name of the variable e.g. MY_VAR
 * @param value value that will be converted to a string and assigned
 */
export function exportVariable(env?: Partial<ENV>, name?: string, value?: string | null | number) {
  if (!env) {
    throw Error(`env must be passed to ${exportVariable.name}`);
  }

  if (!name) {
    throw Error(`env name must be defined got: ${name}`);
  }

  if (!value) {
    console.info(`env ${name} has value ${value}, it will be converted to empty string ''`);
  }

  if (env.TEST_ENV === TEST_ENV.CI || env.TEST_ENV === TEST_ENV.LOCAL) {
    // for test purpose we simply assign
    if (typeof value === 'undefined' || value === null) {
      (env as any)[name] = '';
      return;
    }

    (env as any)[name] = value.toString();
    return;
  }

  // not a test so we execute the assignment with the real utility
  core.exportVariable(name, `${value}`);
}

export function join(paths?: (string | undefined)[]): string {
  if (!paths) {
    return '';
  }

  if (!paths.length) {
    return '';
  }

  return paths.filter((path) => path && path.length > 1).join('.');
}
