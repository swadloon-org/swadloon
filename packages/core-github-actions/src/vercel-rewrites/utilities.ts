import { promises as fsp } from 'fs';

import * as core from '@actions/core';

import { VercelConfig } from './vercel.js';

export function readVercelConfigFile(path: string) {
  core.info(`reading vercel.json file: ${path}`);
  return fsp
    .readFile(path, 'utf-8')
    .then((result) => {
      return JSON.parse(result) as VercelConfig;
    })
    .catch((error) => {
      core.error(`could not read file ${path}`);
      return undefined;
    });
}

export function updateVercelConfigFile(path: string, data: VercelConfig) {
  core.info(`replacing vercel.json file content`);
  return fsp.writeFile(path, JSON.stringify(data, null, 2)).then(() => {
    return readVercelConfigFile(path);
  });
}
