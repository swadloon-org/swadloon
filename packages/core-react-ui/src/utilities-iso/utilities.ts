/**
 *
 * General purposes utility functions
 *
 */

/**
 * Wrapper for Objects.keys
 */
export function keys<T extends object | undefined>(object: T) {
  if (!object) {
    return [] as (keyof T)[];
  }
  return Object.keys(object) as (keyof T)[];
}

/**
 * Wrapper for Objects.values
 */
export function values<T extends object>(object: T) {
  return Object.values(object) as T[keyof T][];
}

/**
 * Convert a number into a px string.
 * @param px value in pixel
 */
export function px({ value }: { value?: number }): string {
  if (!value) {
    return '';
  }
  if (isNaN(value as number)) {
    return '';
  }
  return `${value}px`;
}

/**
 * Convert a number into a em string.
 * @param em value in pixel
 */
export function em({ value }: { value?: number }): string {
  if (!value) {
    return '';
  }
  if (isNaN(value as number)) {
    return '';
  }
  return `${value}em`;
}

/**
 * Convert a number (in px) into a rem value.
 * @param px value in pixel
 * @param baseUnitPx font-size set on the <html/> element
 */
export function pxToRem({ baseUnitPx, value }: { baseUnitPx: number; value: number }): string {
  return `${value / baseUnitPx}rem`;
}

/**
 * Convert a number (in px) into a em relative value.
 * @param px value in pixel
 * @param fontSize font-size set on the element
 */
export function pxToEm({ fontSize, value }: { fontSize: number; value: number }): string {
  return `${value / fontSize}em`;
}

/**
 * Convert px value (e.g. '15px') into a rem value.
 * @param px value in string px value
 * @param baseUnitPx font-size set on the <html/> element
 */
export function pxStringToRem({
  baseFontSize,
  value,
}: {
  baseFontSize: number;
  value: string;
}): string {
  if (!value?.length) {
    return '';
  }

  const pxPattern = /(\d+\.?(\d+))/g;
  const match = pxPattern.exec(value);
  if (!(match?.length && match[0]?.length)) {
    return '';
  }
  const conversion = Number(match[0]);
  if (!conversion) {
    return '';
  }

  return `${conversion / baseFontSize}rem`;
}

/**
 * Convert px value (e.g. '15px') into a number value.
 * @param px value in string px value
 */
export function pxStringToNumber({ value }: { value?: string }): number | undefined {
  if (!value?.length) {
    return undefined;
  }

  const pxPattern = /(\d+\.?(\d+))/g;
  const match = pxPattern.exec(value);
  if (!(match?.length && match[0]?.length)) {
    return undefined;
  }

  const conversion = Number(match[0]);
  if (isNaN(conversion)) {
    return undefined;
  }

  return conversion;
}
