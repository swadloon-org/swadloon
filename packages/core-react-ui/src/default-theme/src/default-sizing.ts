import { SIZE, Sizing, SizingRatios, SizingSteps, VIEWPORT } from '@newrade/core-design-system';

export const defaultBaseFontSize = 9;

export const defaultScalingRatio: SizingRatios = {
  [VIEWPORT.mobile]: 1.44,
  [VIEWPORT.tablet]: 1.44,
  [VIEWPORT.desktop]: 1.618,
};

export const defaultSizingStepsMobile = {
  [SIZE.x0]: 6,
  [SIZE.x1]: 9,
  [SIZE.x2]: 14,
  [SIZE.x3]: 20,
  [SIZE.x4]: 30,
  [SIZE.x5]: 45,
  [SIZE.x6]: 68,
  [SIZE.x7]: 110,
  [SIZE.x8]: 179,
  [SIZE.x9]: 289,
  [SIZE.x10]: 468,
};

export const defaultSizingStepsTablet = {
  [SIZE.x0]: 4,
  [SIZE.x1]: 9,
  [SIZE.x2]: 13,
  [SIZE.x3]: 21,
  [SIZE.x4]: 34,
  [SIZE.x5]: 55,
  [SIZE.x6]: 89,
  [SIZE.x7]: 144,
  [SIZE.x8]: 232,
  [SIZE.x9]: 375,
  [SIZE.x10]: 606,
};

export const defaultSizingStepsDesktop = {
  [SIZE.x0]: 4,
  [SIZE.x1]: 9,
  [SIZE.x2]: 13,
  [SIZE.x3]: 21,
  [SIZE.x4]: 34,
  [SIZE.x5]: 55,
  [SIZE.x6]: 89,
  [SIZE.x7]: 144,
  [SIZE.x8]: 232,
  [SIZE.x9]: 375,
  [SIZE.x10]: 606,
};

export const defaultSizingSteps: SizingSteps = {
  [VIEWPORT.mobile]: defaultSizingStepsMobile,
  [VIEWPORT.tablet]: defaultSizingStepsTablet,
  [VIEWPORT.desktop]: defaultSizingStepsDesktop,
};

export const defaultSizing: Sizing = {
  baseFontSize: defaultBaseFontSize,
  useRatiosForSizes: false,
  ratios: defaultScalingRatio,
  sizes: defaultSizingSteps,
};
