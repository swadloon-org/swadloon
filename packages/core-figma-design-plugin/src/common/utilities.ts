import { pascal } from 'case';

import { VIEWPORT } from '@newrade/core-design-system';
import { keys } from '@newrade/core-react-ui/utilities-iso';

export function formatNameFigma(value?: string | (string | undefined)[]): string {
  if (!value) {
    return '';
  }

  if (Array.isArray(value)) {
    return value.map((str) => (str ? pascal(str) : '')).join('/');
  }

  return pascal(value);
}

export function isViewportProp(value?: VIEWPORT | string): boolean {
  if (!value) {
    return false;
  }

  const found = keys(VIEWPORT).find((viewport) => viewport === value);

  return !!found;
}
