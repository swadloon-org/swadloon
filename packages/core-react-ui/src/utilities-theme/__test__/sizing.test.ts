import { Sizing, VIEWPORT } from '@newrade/core-design-system';

import { defaultSizing } from '../../default-theme/default-sizing.js';
import { getCSSSizingStepV2, getCSSSizingV2 } from '../sizing.utilities.js';

describe('sizing utilities', () => {
  describe(`${getCSSSizingV2.name}`, () => {
    it('should create matching CSS Sizing object', () => {
      const sizing = getCSSSizingV2(defaultSizing) as Sizing<string>;
      expect(sizing).toMatchSnapshot();
    });
  });

  describe(`${getCSSSizingStepV2.name}`, () => {
    it('should create valid CSS sizing steps, given predefined one precise number for each', () => {
      const steps = getCSSSizingStepV2({
        desktop: {
          x0: 4,
          x1: 9,
          x10: 606,
          x2: 13,
          x3: 21,
          x4: 34,
          x5: 55,
          x6: 89,
          x7: 144,
          x8: 232,
          x9: 375,
        },
        mobile: {
          x0: 6,
          x1: 9,
          x10: 468,
          x2: 14,
          x3: 20,
          x4: 30,
          x5: 45,
          x6: 68,
          x7: 110,
          x8: 179,
          x9: 289,
        },
        tablet: {
          x0: 4,
          x1: 9,
          x10: 606,
          x2: 13,
          x3: 21,
          x4: 34,
          x5: 55,
          x6: 89,
          x7: 144,
          x8: 232,
          x9: 375,
        },
      });
      expect(steps).toEqual({
        desktop: {
          x0: '4px',
          x1: '9px',
          x10: '606px',
          x2: '13px',
          x3: '21px',
          x4: '34px',
          x5: '55px',
          x6: '89px',
          x7: '144px',
          x8: '232px',
          x9: '375px',
        },
        mobile: {
          x0: '6px',
          x1: '9px',
          x10: '468px',
          x2: '14px',
          x3: '20px',
          x4: '30px',
          x5: '45px',
          x6: '68px',
          x7: '110px',
          x8: '179px',
          x9: '289px',
        },
        tablet: {
          x0: '4px',
          x1: '9px',
          x10: '606px',
          x2: '13px',
          x3: '21px',
          x4: '34px',
          x5: '55px',
          x6: '89px',
          x7: '144px',
          x8: '232px',
          x9: '375px',
        },
      });
    });

    it('should create valid CSS sizing steps, given modular ratio options', () => {
      const steps = getCSSSizingStepV2(
        {
          desktop: {
            x0: 4,
            x1: 9,
            x2: 13,
            x3: 21,
            x4: 34,
            x5: 55,
            x6: 89,
            x7: 144,
            x8: 232,
            x9: 375,
            x10: 606,
          },
          mobile: {
            x0: 6,
            x1: 9,
            x2: 14,
            x3: 20,
            x4: 30,
            x5: 45,
            x6: 68,
            x7: 110,
            x8: 179,
            x9: 289,
            x10: 468,
          },
          tablet: {
            x0: 4,
            x1: 9,
            x2: 13,
            x3: 21,
            x4: 34,
            x5: 55,
            x6: 89,
            x7: 144,
            x8: 232,
            x9: 375,
            x10: 606,
          },
        },
        {
          [VIEWPORT.desktop]: 1,
          [VIEWPORT.tablet]: 2,
          [VIEWPORT.mobile]: 3,
        }
      );
      expect(steps).toEqual({
        desktop: {
          x0: '4px',
          x1: 'calc(var(--size-x0) * 1 * 1)',
          x2: 'calc(var(--size-x0) * 1 * 1 * 1)',
          x3: 'calc(var(--size-x0) * 1 * 1 * 1 * 1)',
          x4: 'calc(var(--size-x0) * 1 * 1 * 1 * 1 * 1)',
          x5: 'calc(var(--size-x0) * 1 * 1 * 1 * 1 * 1 * 1)',
          x6: 'calc(var(--size-x0) * 1 * 1 * 1 * 1 * 1 * 1 * 1)',
          x7: 'calc(var(--size-x0) * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1)',
          x8: 'calc(var(--size-x0) * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1)',
          x9: 'calc(var(--size-x0) * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1)',
          x10: 'calc(var(--size-x0) * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1 * 1)',
        },
        tablet: {
          x0: '4px',
          x1: 'calc(var(--size-x0) * 2 * 2)',
          x2: 'calc(var(--size-x0) * 2 * 2 * 2)',
          x3: 'calc(var(--size-x0) * 2 * 2 * 2 * 2)',
          x4: 'calc(var(--size-x0) * 2 * 2 * 2 * 2 * 2)',
          x5: 'calc(var(--size-x0) * 2 * 2 * 2 * 2 * 2 * 2)',
          x6: 'calc(var(--size-x0) * 2 * 2 * 2 * 2 * 2 * 2 * 2)',
          x7: 'calc(var(--size-x0) * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2)',
          x8: 'calc(var(--size-x0) * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2)',
          x9: 'calc(var(--size-x0) * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2)',
          x10: 'calc(var(--size-x0) * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2 * 2)',
        },
        mobile: {
          x0: '6px',
          x1: 'calc(var(--size-x0) * 3 * 3)',
          x2: 'calc(var(--size-x0) * 3 * 3 * 3)',
          x3: 'calc(var(--size-x0) * 3 * 3 * 3 * 3)',
          x4: 'calc(var(--size-x0) * 3 * 3 * 3 * 3 * 3)',
          x5: 'calc(var(--size-x0) * 3 * 3 * 3 * 3 * 3 * 3)',
          x6: 'calc(var(--size-x0) * 3 * 3 * 3 * 3 * 3 * 3 * 3)',
          x7: 'calc(var(--size-x0) * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3)',
          x8: 'calc(var(--size-x0) * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3)',
          x9: 'calc(var(--size-x0) * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3)',
          x10: 'calc(var(--size-x0) * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3 * 3)',
        },
      });
    });
  });
});
