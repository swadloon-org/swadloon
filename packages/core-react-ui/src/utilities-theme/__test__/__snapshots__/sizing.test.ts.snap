// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sizing utilities getCSSSizingV2 should create matching CSS Sizing object 1`] = `
Object {
  "baseFontSize": "9px",
  "ratios": Object {
    "desktop": "1.618",
    "mobile": "1.44",
    "tablet": "1.44",
  },
  "sizes": Object {
    "desktop": Object {
      "x0": "4px",
      "x1": "9px",
      "x10": "606px",
      "x2": "13px",
      "x3": "21px",
      "x4": "34px",
      "x5": "55px",
      "x6": "89px",
      "x7": "144px",
      "x8": "232px",
      "x9": "375px",
    },
    "mobile": Object {
      "x0": "6px",
      "x1": "9px",
      "x10": "468px",
      "x2": "14px",
      "x3": "20px",
      "x4": "30px",
      "x5": "45px",
      "x6": "68px",
      "x7": "110px",
      "x8": "179px",
      "x9": "289px",
    },
    "tablet": Object {
      "x0": "4px",
      "x1": "9px",
      "x10": "606px",
      "x2": "13px",
      "x3": "21px",
      "x4": "34px",
      "x5": "55px",
      "x6": "89px",
      "x7": "144px",
      "x8": "232px",
      "x9": "375px",
    },
  },
  "useRatiosForSizes": "false",
}
`;
